ecosystem/sep-0010.md

```diff
@@ -139,7 +139,75 @@ operation of the challenge is the **Home Domain**, and that the web auth domain
 GET <WEB_AUTH_ENDPOINT>
 ```
 
-Request Parameters:
+##### Request Header:
+
+Optionally, the server may forbid unauthorized calls to `GET <WEB_AUTH_ENDPOINT>`. This could be done to protect an
+endpoint against bad actors or limit access only to allowed applications.
+
+In that case, the client must add an `Authorization` header, and the server should respond to all requests with missing
+header with an error (see below). Server may choose to only accept calls from trusted applications, responding with an
+error to all other requests (see below).
+
+Authorized `WEB_AUTH_ENDPOINT` accepts `Authorization Bearer: <token>`, where token is [JSON Web Token](https://jwt.io/)
+authorizing the request.
+
+Client must specify the following claims:
+
+- `iat` (the time at which the JWT was issued
+  [RFC7519, Section 4.1.6](https://tools.ietf.org/html/rfc7519#section-4.1.6)) — current timestamp (`1530644093`)
+- `exp` (the expiration time on or after which the JWT must not be accepted for processing,
+  [RFC7519, Section 4.1.4](https://tools.ietf.org/html/rfc7519#section-4.1.4)) — a server can pick its own expiration
+  period for the token (`1530730493`)
+- `url` full URL of the request, including scheme, path and query parameters.
+
+Client must then correctly sign the payload with appropriate Stellar private key. To choose the private key client
+application should follow this steps:
+
+- If `client_domain` is specified, the token must be signed with the `SIGNING_KEY`. (See
+  [Verifying the Client Domain](#verifying-the-client-domain) section for more information)
+- Otherwise, the token must be signed with a private key of the **Client Account** passed in the `account` field.
+
+Token then should be signed with the selected private key, using **ed25519** algorithm. An appropriate header must be
+included
+
+```json
+{
+  "alg": "EdDSA"
+}
+```
+
+Example of the valid signed token in base64 format:
+
+```
+eyJhbGciOiJFZERTQSJ9.eyJqdGkiOiIzYjExYTRhYS1lMmE4LTRkYTUtYTUxOC1iMDU5YWMwMjUwOGYiLCJpYXQiOjE3MTA5NzY2MDcsImV4cCI6MTAzNTA5NzY2MDcsInVybCI6Imh0dHBzOi8vYXV0aC5leGFtcGxlLmNvbS8_YWNjb3VudD1HQ0lCVUNHUE9IV01NTUZQRlREV0JTVkhRUlQ0RElCSjdBRDZCWkpZRElUQksyTENWQllXN0hVUSZjbGllbnRfZG9tYWluPWV4YW1wbGUtd2FsbGV0LnN0ZWxsYXIub3JnIn0.FhX1sxGZcFMEwfRE4W3fe8uB5PUtBBS3_nkKUVHx9Ulw3Kc3OvJ5bwLRXuWRpBndvuK2S-yKdKBUCLqd4QO-Cg
+```
+
+This token was signed with the `GCADAIACE6CRWGOB3HJRXIOHMQEUKPNLAUIYFIE26F3OK72RSHGAMARK` key, belonging to a
+hypothetical domain `example-wallet.stellar.org`
+
+Another example of the valid signed token:
+
+```
+eyJhbGciOiJFZERTQSJ9.eyJqdGkiOiI1YWExYTNhYi0zY2Q2LTRkYWYtODQyMi0xYmI5Zjc1NTI1YjEiLCJpYXQiOjE3MTA5NzY5MDYsImV4cCI6MTAzNTA5NzY5MDYsInVybCI6Imh0dHBzOi8vYXV0aC5leGFtcGxlLmNvbS8_YWNjb3VudD1HQ1hYSDZBWUpVVlRER0lIVDQyT1pOTUYzTEhDVjRET0tDWDZISERLV0VDVVpZWERaU1daTjZIUyZtZW1vPTEyMzQ1NjcifQ.Gi7aPIwLBss54DJzKhi9lRlUViCkLbbjT0SQFrG9eGGt7TAGD1VeLfj2LFBCOoCFytXWDufplY4iJk4XG_jwDw
+```
```

Comment:
This token has a `jti` property that is not defined in this standard

Reply:
I was thinking about if we should add it or not. I think we should allow clients add their own fields or maybe add JTI as optional? IMO it can be useful

Reply:
I don't feel strongly either way, but my inclination is to only add fields that satisfy a requirement of what we're trying to solve.